tests/models/test_tpu.py

```diff
@@ -495,3 +495,24 @@ def teardown(self, stage):
 
     model = DebugModel()
     tpipes.run_model_test(trainer_options, model, on_gpu=False, with_hpc=False)
+
+
+@RunIf(tpu=True)
+@pl_multi_process_test
+def test_predict_tpu_multi_cores(tmpdir):
```

Comment:
I think an unit test would be better suited for this PR

Reply:
What do you have in mind? Just check the `len` attribute for `IndexBatchSamplerWrapper`?

Reply:
Yeah, simple and effective 😜

Reply:
Done!